programs/MouseDemo.hs

```diff
@@ -51,6 +56,18 @@ buttonLayer st =
                padLeftRight (if wasClicked then 2 else 3) $
                str (if wasClicked then "<" <> label <> ">" else label)
 
+editLayer :: St -> Widget Name
+editLayer st =
+  B.border $
+  C.hCenterLayer $
+  vLimit 8 $
+  -- n.b. if clickable and viewport are inverted here, click event
+  -- coordinates will only identify the viewable range, not the actual
+  -- editor widget coordinates.
```

Comment:
Yes, the editor is probably a confusing example for that reason.  I should (and probably will) change this to a viewport of a vbox of str widgets which is better.

With the current code, if you invert the next two lines, the click events are reported for the viewport widget wrapper (i.e. the row is always 0-7, regardless of the scrolling of the viewport contents).  In the order shown here, if the viewport contents are scrolled down by 3 lines then the click reporting row will be in the range 3-10.

I will check what the functionality is if the EditedView viewport is removed in favor of the Editor's internal viewport.